lib/ClientImpl.cc

```diff
@@ -497,6 +497,20 @@ void ClientImpl::getPartitionsForTopicAsync(const std::string& topic, GetPartiti
                   std::placeholders::_2, topicName, callback));
 }
 
+Result ClientImpl::updateServiceUrl(const std::string& serviceUrl) {
+    LOG_INFO("Updating service URL to " << serviceUrl);
+
+    try {
+        serviceNameResolver_.updateServiceUrl(serviceUrl);
+        lookupServicePtr_->updateServiceUrl(serviceUrl);
```

Comment:
Actually we don't need to add `updateServiceUrl` method to `LookupService` because in C++ client, all `LookupService` implementations just call  `serviceNameResolver_.updateServiceUrl(serviceUrl);`.

Unlike Java client, `HttpLookupService` uses a `HttpClient` class that accepts a service URL.

Reply:
Ok, I will remove `updateServiceUrl`  method of `LookupService `.